### PR TITLE
chore(scheduler): add tracing

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes

Add tracing to the scheduler package

Partially addressing https://linear.app/nango/issue/NAN-1224/improve-orchestrator-monitoring

Tested in staging

### CODE REVIEW

File: packages/scheduler/lib/workers/monitor/monitor.worker.boot.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { isMainThread, parentPort, workerData } from 'node:worker_threads';
 import { MonitorChild } from './monitor.worker.js';
 import { DatabaseClient } from '../../db/client.js';
 import { logger } from '../../utils/logger.js';
+import '../../tracer.js';
```
Comment (bodinsamuel) [inline packages/scheduler/lib/workers/monitor/monitor.worker.boot.ts]: should be at the beginning

File: packages/scheduler/lib/workers/scheduling/scheduling.worker.boot.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { isMainThread, parentPort, workerData } from 'node:worker_threads';
 import { SchedulingChild } from './scheduling.worker.js';
 import { DatabaseClient } from '../../db/client.js';
 import { logger } from '../../utils/logger.js';
+import '../../tracer.js';
```
Comment (bodinsamuel) [inline packages/scheduler/lib/workers/scheduling/scheduling.worker.boot.ts]: same

File: packages/scheduler/lib/workers/scheduling/scheduling.worker.ts [hunk 3]
```diff
@@ -95,49 +96,68 @@ export class SchedulingChild {
...
-                    );
-                    const res = await Promise.allSettled(createTasks);
-                    for (const taskRes of res) {
-                        if (taskRes.status === 'rejected') {
-                            logger.error(`Failed to schedule task: ${taskRes.reason}`);
-                        } else if (taskRes.value.isErr()) {
-                            logger.error(`Failed to schedule task: ${stringifyError(taskRes.value.error)}`);
-                        } else {
-                            taskIds.push(taskRes.value.value.id);
+            const span = tracer.startSpan('scheduler.scheduling.schedule');
```
Comment (bodinsamuel) [inline packages/scheduler/lib/workers/scheduling/scheduling.worker.ts:99]: note if you don't wrap in a callback the next calls will not be wrapped in this span

**Review by bodinsamuel — APPROVED** [review]
Minor comments

File: packages/scheduler/lib/workers/scheduling/scheduling.worker.ts (hunk 3 above)
Comment (TBonnin) [reply packages/scheduler/lib/workers/scheduling/scheduling.worker.ts:99]: fixed in https://github.com/NangoHQ/nango/pull/2374/commits/a5d1699e88a4933329438cf3abc5160ac36e6e80


Looks nicer now !
[Screenshot 2024-06-20 at 09 53 54](https://github.com/NangoHQ/nango/assets/233326/fb0c534b-eb3f-457d-95b8-3d1dd2affb60)

**Review by bodinsamuel — APPROVED** [review]
👣

**Pull request merged**